pkgs/build-support/coq/default.nix

```diff
@@ -4,106 +4,112 @@ with builtins; with lib;
+    useDune2 = args.useDune2 or useDune2ifVersion fetched.version;
+  in
 
-stdenv.mkDerivation (removeAttrs ({
+  flip removeAttrs dropDerivationAttrs ({
 
-  name = prefix-name + (display-pkg pname "-" fetched.version);
+    name = prefix-name + (display-pkg pname "-" fetched.version);
 
-  inherit (fetched) version src;
+    inherit (fetched) version src;
 
-  buildInputs = [ coq ]
-    ++ optionals mlPlugin coq.ocamlBuildInputs
-    ++ optionals useDune2 [coq.ocaml coq.ocamlPackages.dune_2]
-    ++ extraBuildInputs;
-  inherit enableParallelBuilding;
+    buildInputs = [ coq ]
+      ++ optionals mlPlugin coq.ocamlBuildInputs
+      ++ optionals useDune2 [coq.ocaml coq.ocamlPackages.dune_2]
+      ++ extraBuildInputs;
+    inherit enableParallelBuilding;
 
-  meta = ({ platforms = coq.meta.platforms; } //
-    (switch domain [{
-        case = pred.union isGitHubDomain isGitLabDomain;
-        out = { homepage = "https://${domain}/${owner}/${repo}"; };
-      }] {}) //
-    optionalAttrs (fetched.broken or false) { coqFilter = true; broken = true; }) //
-    (args.meta or {}) ;
+    meta = ({ platforms = coq.meta.platforms; } //
+      (switch domain [{
+          case = pred.union isGitHubDomain isGitLabDomain;
+          out = { homepage = "https://${domain}/${owner}/${repo}"; };
+        }] {}) //
+      optionalAttrs (fetched.broken or false) { coqFilter = true; broken = true; }) //
+      (args.meta or {}) ;
 
-}
-// (optionalAttrs setCOQBIN { COQBIN = "${coq}/bin/"; })
-// (optionalAttrs (!args?installPhase && !args?useMelquiondRemake) {
-  installFlags =
-    [ "${var-coqlib-install}=$(out)/lib/coq/${coq.coq-version}/" ] ++
-    optional (match ".*doc$" (args.installTargets or "") != null)
-      "DOCDIR=$(out)/share/coq/${coq.coq-version}/" ++
-    extraInstallFlags;
-})
-// (optionalAttrs useDune2 {
-  installPhase = ''
-    runHook preInstall
-    dune install --prefix=$out
-    mv $out/lib/coq $out/lib/TEMPORARY
-    mkdir $out/lib/coq/
-    mv $out/lib/TEMPORARY $out/lib/coq/${coq.coq-version}
-    runHook postInstall
-  '';
-})
-// (optionalAttrs (args?useMelquiondRemake) rec {
-  COQUSERCONTRIB = "$out/lib/coq/${coq.coq-version}/user-contrib";
-  preConfigurePhases = "autoconf";
-  configureFlags = [ "--libdir=${COQUSERCONTRIB}/${useMelquiondRemake.logpath or ""}" ];
-  buildPhase = "./remake -j$NIX_BUILD_CORES";
-  installPhase = "./remake install";
-})
-// (removeAttrs args args-to-remove)) dropDerivationAttrs)
+  }
+  // (optionalAttrs setCOQBIN { COQBIN = "${coq}/bin/"; })
+  // (optionalAttrs (!args?installPhase && !args?useMelquiondRemake) {
+    installFlags =
+      [ "${var-coqlib-install}=$(out)/lib/coq/${coq.coq-version}/" ] ++
+      optional (match ".*doc$" (args.installTargets or "") != null)
+        "DOCDIR=$(out)/share/coq/${coq.coq-version}/" ++
+      extraInstallFlags;
+  })
+  // (optionalAttrs useDune2 {
+    installPhase = ''
+      runHook preInstall
+      dune install --prefix=$out
```

Comment:
This is unrelated to this PR, but this is actually incorrect in case of a monorepo. Anyway, `useDune2` is also broken for other reasons (it doesn't override the configure and build phases).

Reply:
This was fixed in another PR.